dot/state/block.go

```diff
@@ -29,13 +29,14 @@ const (
 )
 
 var (
-	headerPrefix        = []byte("hdr") // headerPrefix + hash -> header
-	blockBodyPrefix     = []byte("blb") // blockBodyPrefix + hash -> body
-	headerHashPrefix    = []byte("hsh") // headerHashPrefix + encodedBlockNum -> hash
-	arrivalTimePrefix   = []byte("arr") // arrivalTimePrefix || hash -> arrivalTime
-	receiptPrefix       = []byte("rcp") // receiptPrefix + hash -> receipt
-	messageQueuePrefix  = []byte("mqp") // messageQueuePrefix + hash -> message queue
-	justificationPrefix = []byte("jcp") // justificationPrefix + hash -> justification
+	headerPrefix          = []byte("hdr") // headerPrefix + hash -> header
+	blockBodyPrefix       = []byte("blb") // blockBodyPrefix + hash -> body
+	headerHashPrefix      = []byte("hsh") // headerHashPrefix + encodedBlockNum -> hash
+	arrivalTimePrefix     = []byte("arr") // arrivalTimePrefix || hash -> arrivalTime
+	receiptPrefix         = []byte("rcp") // receiptPrefix + hash -> receipt
+	messageQueuePrefix    = []byte("mqp") // messageQueuePrefix + hash -> message queue
+	justificationPrefix   = []byte("jcp") // justificationPrefix + hash -> justification
+	firstSlotNumberPrefix = []byte("fsn") // firstSlotNumberPrefix + hash -> first slot number
```

Comment:
you can leave it as just the `[]byte(fsn)` since this info is unique in the chain